runtime/j9vm/java11vmi.c

```diff
@@ -1458,28 +1458,39 @@ JVM_GetVmArguments(JNIEnv *env) {
 	J9VMThread* currentThread = (J9VMThread*)env;
 	J9JavaVM* vm = currentThread->javaVM;
 	J9InternalVMFunctions* internalFunctions = vm->internalVMFunctions;
-	jclass vmJniClass = NULL;
 	jobjectArray result = NULL;
-	J9Class* vmClass = J9VMCOMIBMOTIVMVM_OR_NULL(vm);
+	J9Class* vmClass = NULL;
 
-	if (NULL == vmClass) {
-		internalFunctions->setCurrentException(currentThread, J9VMCONSTANTPOOL_JAVALANGINTERNALERROR, NULL);
-	} else {
-		jclass vmJniClass = NULL;
-		jmethodID mid = NULL;
+	internalFunctions->internalEnterVMFromJNI(currentThread);
 
-		vmJniClass = (jclass)internalFunctions->j9jni_createLocalRef(env, vmClass->classObject);
+	vmClass = J9VMCOMIBMOTIVMVM_OR_NULL(vm);
 
-		mid = (*env)->GetStaticMethodID(env, vmJniClass, "getVMArgs", "()[Ljava/lang/String;");
-		if (NULL == mid) {
-			internalFunctions->setCurrentException(currentThread, J9VMCONSTANTPOOL_JAVALANGINTERNALERROR, NULL);
-		} else {
-			result = (jobjectArray)((*env)->CallObjectMethod(env, vmJniClass, mid));
+	if (NULL != vmClass) {
+		J9Method* method = method = internalFunctions->findJNIMethod(currentThread, vmClass, "getVMArgs", "()[Ljava/lang/String;");
+
+		if (NULL != method) {
+			jmethodID mid = (jmethodID)internalFunctions->getJNIMethodID(currentThread, method);
+
+			if (NULL != mid) {
+				jclass vmJniClass = vmJniClass = (jclass)internalFunctions->j9jni_createLocalRef(env, vmClass->classObject);
```

Comment:
`jclass vmJniClass = vmJniClass = `

Reply:
This can fail - it needs a nullcheck / exception check. You'll have to look at j9jni_createLocalRef() to see what the contract is on failure.